src/shared/lib/resources/opportunity/sprint-with-us.ts

```diff
@@ -0,0 +1,217 @@
+  reporting?: {
+    numProposals: number;
+    numWatchers: number;
+    numViews: number;
+  };
+}
+
+export interface SWUOpportunityPhase {
+  startDate: Date;
+  completionDate: Date;
+  maxBudget: number;
+  createdAt: Date;
+  createdBy: UserSlim;
+  updatedAt: Date;
+  updatedBy: UserSlim;
+  requiredCapabilities: string[];
+}
+
+export interface SWUTeamQuestion {
+  question: string;
+  guideline: string;
+  score: number;
+  wordLimit: number;
+  order: number;
+  createdAt: Date;
+  createdBy: UserSlim;
+  updatedAt: Date;
+  updatedBy: UserSlim;
+}
+
+export type SWUOpportunitySlim = Pick<SWUOpportunity, 'id' | 'title' | 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy' | 'status' | 'proposalDeadline'>;
+
+// Create.
+
+export type CreateSWUOpportunityStatus
+  = SWUOpportunityStatus.Published
+  | SWUOpportunityStatus.Draft;
+
+export interface CreateSWUOpportunityPhaseBody extends Omit<SWUOpportunityPhase, 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy' | 'startDate' | 'completionDate'> {
+  startDate: string;
+  completionDate: string;
+}
+
+export type CreateSWUTeamQuestionBody = Omit<SWUTeamQuestion, 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy'>;
+
+export interface CreateRequestBody {
+  title: string;
+  teaser: string;
+  remoteOk: boolean;
+  remoteDesc: string;
+  location: string;
+  totalMaxBudget: number;
+  minTeamMembers: number;
+  mandatorySkills: string[];
+  optionalSkills: string[];
+  description: string;
+  proposalDeadline: string;
+  assignmentDate: string;
+  questionsWeight: number;
+  codeChallengeWeight: number;
+  scenarioWeight: number;
+  priceWeight: number;
+  attachments: Id[];
+  status: CreateSWUOpportunityStatus;
+  inceptionPhase?: CreateSWUOpportunityPhaseBody;
+  prototypePhase?: CreateSWUOpportunityPhaseBody;
+  implementationPhase: CreateSWUOpportunityPhaseBody;
+  teamQuestions: CreateSWUTeamQuestionBody[];
+}
+
+export interface CreateSWUOpportunityPhaseValidationErrors extends Omit<ErrorTypeFrom<CreateSWUOpportunityPhaseBody>, 'requiredCapabilities'> {
+  requiredCapabilities?: string[][];
+}
+
+export interface CreateSWUTeamQuestionValidationErrors extends ErrorTypeFrom<CreateSWUTeamQuestionBody> {
+  parseFailure?: string[];
+}
+
+export interface CreateValidationErrors extends Omit<ErrorTypeFrom<CreateRequestBody> & BodyWithErrors, 'mandatorySkills' | 'optionalSkills' | 'inceptionPhase' | 'prototypePhase' | 'implementationPhase' | 'teamQuestions' | 'attachments'> {
+  mandatorySkills?: string[][];
+  optionalSkills?: string[][];
+  inceptionPhase?: CreateSWUOpportunityPhaseValidationErrors;
+  prototypePhase?: CreateSWUOpportunityPhaseValidationErrors;
+  implementationPhase?: CreateSWUOpportunityPhaseValidationErrors;
+  teamQuestions?: CreateSWUTeamQuestionValidationErrors[];
+  attachments?: string[][];
+}
+
+// Update.
+
+export type UpdateRequestBody
+  = ADT<'edit', UpdateEditRequestBody>
+  | ADT<'publish', string>
+  | ADT<'evaluateQuestions', string>
```

Comment:
`evaluateQuestions` would only be called automatically by a hook when a proposal deadline lapses — no need to have it as part of the update request body.

Reply:
Removed, and added 'submitForReview' update request body.